fix(gux-table): use percentage widths for resizable columns to alter …

thomas-c-dillon opened the pull request:
…behavior when resizing viewport

COMUI-1190

Move from static pixel widths on resizable table columns to percentage widths so that all columns scale with viewport width change (instead of only the last column).

MattCheely commented on src/components/beta/gux-table/gux-table.tsx:
```diff
@@ -399,34 +399,43 @@ export class GuxTable {
   }
 
-  /******************************* Resizeable Columns *******************************/
+  /******************************* Resizable Columns *******************************/
```
Thank you for fixing all  of my typos.

MattCheely commented on src/components/beta/gux-table/gux-table.tsx:
```diff
@@ -470,12 +479,36 @@ export class GuxTable {
+
+    return columns.map((c, _, arr) => ({
+      ...c,
+      width: ((c.width / calcTableWidth(arr)) * 100).toFixed(2) + '%'
```
Using the array inside the map is a bit confusing and creates unnecessary repeat calculations. I also find the math and string concatenation inside the spread assignment inside the anonymous function a bit hard to parse. 

I'd maybe add a helper function/method:

```typescript
pixelsToCSSPercent(section, total) {
  return (section / total) * 100).toFixed(2) + '%'
}
```

And then make the body of this function:

```typescript
const totalPxWidth = columns.reduce((total, col) => total + col.width, 0);

return columns.map((c) => ({
  ...c,
  width: pixelsToPercent(c.width, totalWidth)
}));
```
IMO, that makes things easier to read and a bit more self documenting.

thomas-c-dillon replied:
Great improvement. I didn't want to go too far down the rabbit hole of refactoring (never know when to stop when I start).

MattCheely reviewed (changes requested):
This solution is quite nice. There's just a few spots where I think the implementation will be hard to parse for maintainers like me with brains burnt out on sleep deprivation and lots of planning meetings. 😅

thomas-c-dillon commented on src/components/beta/gux-table/gux-table.tsx:
```diff
@@ -399,34 +399,43 @@ export class GuxTable {
-    });
+    this.calculateColumnPercentWidth(columnWidths)
+      // Exclude the last column to allow it to fill the remaining space naturally
+      .filter((_c, index, arr) => index !== arr.length - 1)
```
I have no preference, works for me!

thomas-c-dillon commented:
I have unfortunately just noticed a major issue with these changes. Looks like we are unable to resize columns to greater than 100% of the table width. I welcome any ideas on that issue, otherwise I am going have to dig back in tomorrow.

thomas-c-dillon commented:
Reverted resizable columns to use static pixel widths to allow the table to overflow beyond 100% width. Instead, added an event handler for the resize observer to scale columns (keeping the same percentage width) when container is resized.
